examples/nidaqmx/analog-input.py

```diff
@@ -73,10 +72,16 @@ def RaiseIfError(response):
 
     RaiseIfError(client.StartTask(nidaqmx_types.StartTaskRequest(task=task)))
 
```

Comment:
@harsha-bhushan @gregstoll I tried to think of something for the basic examples that is easy to understand. Hopefully someone can look at this and understand (a) how attribute accesses work and (b) what this particular attribute access does and whether they would need to use it in their program.

I also don't want to add a lot of noise to these examples by reading/writing a bunch of attributes.

I can't think of good examples for setting attributes. The Buffer attributes are easy ones that might be useful. But if this were an example for a true beginner, they might think that setting those attributes is recommended or necessary for simple acquisition (it's not). We could set many of the attributes that are part of the initial config of the sample clock or the channel (AI_MAX, channel name, Rate, etc). 

That's redundant but mostly harmless. Some of the Read attrbutes would be reasonable:
```cpp
DAQmxSetReadAutoStart
DAQmxSetReadOverWrite
```
But those have the same problem as the buffer attributes.

Greg's advanced example has a lot of attribute accesses.
We could also consider an example *just* for accessing attributes that doesn't actually do any acquisition.

Reply:
Sorry if my email was confusing. 
As long as any of the shipping example contains accessing attributes, that should be good. Yes, I agree with your point that we should not forcefully add attributes where it is not needed as that can misguide new users. If the advanced examples are covering it that is good enough, we need not have to put them in these examples or create an example just to showcase attribute access.

Reply:
We anyway plan to add some documentation on generic attribute get/set in the DAQmx page of the wiki. we can include this scenario too & link to the example that contains the attribute access. Since we have quite a few things to document, it might be good to have one section on 'attributes' in the DAQmx page of the wiki instead of putting it under an already existing section.

Reply:
You could revert back this change as I looked at the triggering example that nicely covers the usage of get/set attributes. Between that example & documentation users should be able to understand the usage. We can keep the analog input simple.

Reply:
The more I'm looking at the usage here, the more I'm feeling we should not be doing this :p 
Just thinking out loud a few options of creating genuine scenarios:
1.Tweak one of the examples to have multiple channels & then use this way to determine the array? (every-n-samples example or the synchronous input output example that we are yet to create) 
2.Duplicate this example but have multiple channels & name it analog-input-multi-channel & show this usage of setting array size
Any other options too. We can pick whatever is simple & quick.

Reply:
Hah. This change actually has a use case *in this script*, I just didn't realize it. You can pass in a "list or range of channels" to the create channel APIs!

With my slight change to the `print` out in the latest update:

```
(.venv) λ python analog-input.py 10.0.62.17 31763 gRPCSystemTestDAQ/ai0:3
Acquired 4000 samples (1000 samples per channel)
```

Reply:
PS: thanks for taking the time to provide detailed feedback!

Reply:
great. then it would be worth adding this in other examples too. and in line 20 - code documentation, we can add "to acquire from 'n' consecutive channels, pass 'Dev1/ai0:n' as parameter". you could rephrase to something better.

What is your thought on this?

Reply:
Good idea.

The standard phrasing in the [C API docs ](https://zone.ni.com/reference/en-XX/help/370471AM-01/daqmxcfunc/daqmxcreateaivoltagechan/) is "The names of the physical channels to use to create virtual channels. You can specify a [list or range](https://zone.ni.com/reference/en-XX/help/370466AH-01/mxcncpts/physchannames/) of physical channels." There are several different valid syntaxes.

I think I will at least use the "list or range" terminology and give an example like "i.e., Dev/ai0:n specifies a range of channels from 0 to n".

If we can actually link the help concept, that would be great. I don't know if those links are stable (and ofcourse we're expecting the help to change soon ...)

Reply:
I updated all of the analog input examples. With digital input, you often end up with only one "channel" even when you pass in a list/range of inputs. So, the same mechanism can be used, but it's not as intuitive or as common. I don't want to make those examples more complicated or potentially misleading.